src/observer.ts

```diff
@@ -362,21 +398,26 @@ export class DataLayerObserver {
 
     try {
       const target = DataLayerTarget.find(source);
-      this.registerTarget(target, operators, destination, readOnLoad, monitor, debug, debounce);
-    } catch (_) {
-      // schedule subsequent attempts at (attempt * wait) later
-      setTimeout(() => {
-        try {
-          const target = DataLayerTarget.find(source);
-          this.registerTarget(target, operators, destination, readOnLoad, monitor, debug, debounce);
-        } catch (err) {
-          if (attempt > 3) {
-            Logger.getInstance().error(LogMessageType.RuleRegistrationError, { rule: id, source, reason: err.message });
+
+      switch (typeof waitUntil) {
```

Comment:
Internally the `registerRule` defaults to a `waitUntil` function that checks `undefined` as well as the data layer is of the supported types.